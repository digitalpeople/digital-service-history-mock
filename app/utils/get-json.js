import fs from 'fs';
import responseHandler from './response-handler';

const getJson = (path, response) => {
  response.append('Content-Type', 'application/json');
  fs.readFile(`${path}`, responseHandler(response));
};

export default getJson;
