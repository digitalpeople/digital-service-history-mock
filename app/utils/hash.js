import bcrypt from 'bcrypt';

const hash = password => bcrypt.hash(password, 10);
const compare = (actual, expected) => bcrypt.compare(actual, expected);

export { hash, compare };
