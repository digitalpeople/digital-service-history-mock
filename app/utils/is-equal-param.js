const isEqualParam = (request, path, value) => (
  typeof path === 'string' && ('params' in request) ?
    request.params[path] === value :
    false
);

export default isEqualParam;
