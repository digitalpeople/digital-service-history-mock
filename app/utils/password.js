import bcrypt from 'bcrypt';

const hashPassword = password => bcrypt.hash(password, 10);
const comparePassword = (actual, expected) => bcrypt.compare(actual, expected);

export { hashPassword, comparePassword };
