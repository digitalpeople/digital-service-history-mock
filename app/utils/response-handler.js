const responseHandler = response => (error, data) => {
  if (error) {
    response.sendStatus(404);
    return;
  }

  response.send(data);
};

export default responseHandler;
