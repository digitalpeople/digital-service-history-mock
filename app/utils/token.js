import jwt from 'jsonwebtoken';
import jwtKey from '../../data/mock-jwt-key';

const createToken = content => jwt.sign(content, jwtKey, { expiresIn: '1h' });

export default createToken;
