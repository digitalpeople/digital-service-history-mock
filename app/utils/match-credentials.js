import { compare } from './hash';
import {
  mockUsername,
  mockPassword,
} from '../../data/mock-credentials';

const matchCredentials = async (username, password) => {
  const usernameMatch = username === mockUsername;
  const passwordMatch = await compare(password, await mockPassword);

  return usernameMatch && passwordMatch;
};

export default matchCredentials;
