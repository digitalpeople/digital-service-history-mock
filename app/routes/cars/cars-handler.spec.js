import {
  getCarsHandler,
  getCarsByIdHandler,
} from './cars-handler';

import * as getJson from '../../utils/get-json';

describe('getCarsHandler', () => {
  it('should call getJson with correct path', () => {
    expect.assertions(1);

    getJson.default = jest.fn();
    const request = {
      foo: 'baz',
    };
    const response = {
      foo: 'bar',
    };

    getCarsHandler(request, response);

    expect(getJson.default.mock.calls[0]).toMatchSnapshot();
  });
});

describe('getCarsByIdHandler', () => {
  it('should call getJson with correct path', () => {
    expect.assertions(1);

    getJson.default = jest.fn();
    const request = {
      foo: 'baz',
    };
    const response = {
      foo: 'bar',
    };

    getCarsByIdHandler(request, response);

    expect(getJson.default.mock.calls[0]).toMatchSnapshot();
  });
});
