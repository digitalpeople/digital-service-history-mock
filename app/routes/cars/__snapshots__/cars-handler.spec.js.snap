// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCarsByIdHandler should call getJson with correct path 1`] = `
Array [
  "data/cars-by-id.json",
  Object {
    "foo": "bar",
  },
]
`;

exports[`getCarsHandler should call getJson with correct path 1`] = `
Array [
  "data/cars.json",
  Object {
    "foo": "bar",
  },
]
`;
